akka-typed/src/main/scala/akka/typed/persistence/scaladsl/PersistentActor.scala

```diff
@@ -21,18 +21,27 @@ object PersistentActor {
 
   sealed abstract class PersistentEffect[+Event, State]() {
     def andThen(callback: State ⇒ Unit): PersistentEffect[Event, State]
+    def andThen(callback: ⇒ Unit): PersistentEffect[Event, State]
   }
 
   final case class PersistNothing[Event, State](callbacks: List[State ⇒ Unit] = Nil) extends PersistentEffect[Event, State] {
     def andThen(callback: State ⇒ Unit) = copy(callbacks = callback :: callbacks)
+    def andThen(callback: ⇒ Unit) = copy(callbacks = ((_: State) ⇒ callback) :: callbacks)
   }
 
-  case class Persist[Event, State](event: Event, callbacks: List[State ⇒ Unit] = Nil) extends PersistentEffect[Event, State] {
+  case class PersistWithCallback[Event, State](event: Event, callbacks: List[State ⇒ Unit]) extends PersistentEffect[Event, State] {
```

Comment:
I understand the purpose of the two `andThen` but why is this class needed?
It wasn't needed for `Unhandled` for example.

Wont this be superseded by the composite effects anyway?

Reply:
indeed!